src/math/PercentageMath.sol

```diff
@@ -100,45 +99,41 @@ library PercentageMath {
         }
     }
 
-    /// @notice Executes a weighted average, given an interval [x, y] and a percent p: x * (1 - p) + y * p
-    /// @param x The value at the start of the interval (included).
-    /// @param y The value at the end of the interval (included).
-    /// @param percentage The percentage of the interval to be calculated.
-    /// @return z The average of x and y, weighted by percentage.
+    /// @notice Executes a weighted average (x * (1 - p) + y * p), rounded up.
+    /// @param x The first value, with a weight of 1 - percentage.
+    /// @param y The second value, with a weight of percentage.
+    /// @param percentage The weight of y, and complement of the weight of x.
+    /// @return z The result of the weighted average.
     function weightedAvg(
         uint256 x,
         uint256 y,
         uint256 percentage
     ) internal pure returns (uint256 z) {
-        // Let percentage > 0
-        // Underflow if percentage > PERCENTAGE_FACTOR
-        //
-        // Overflow if x1 = x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
-        // <=> x * (PERCENTAGE_FACTOR - percentage) > type(uint256).max - HALF_PERCENTAGE_FACTOR
-        // <=> x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
-        //
-        // Overflow if x2 = y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
+        // Must revert if
+        // percentage > PERCENTAGE_FACTOR
+        // or
+        // x1 = x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
+        // <=> (PERCENTAGE_FACTOR - percentage) > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
+        // or
+        // x2 = y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
         // <=> y * percentage > type(uint256).max - HALF_PERCENTAGE_FACTOR
-        // <=> y > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / percentage
+        // <=> percentage > 0 and y > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / percentage
         //
```

Comment:
It is really difficult to parse the order of the logical operators here. We can be more explicit and add a "Must also revert if" in between

Reply:
Perhaps we can keep the previous version, or tweak it

Reply:
Indentation seems to be a good idea here